roles/openshift_metering/tasks/install.yaml

```diff
@@ -0,0 +1,114 @@
+---
+- name: Create temp directory for doing work in on target
+  command: mktemp -td openshift-metering-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+
+- name: Ensures {{ mktemp.stdout }}/metering-manifests dir exists
+  file:
+    path: "{{ mktemp.stdout }}/metering-manifests"
+    state: directory
+
+- name: Copy files to temp directory
+  copy:
+    src: "{{ item }}"
+    dest: "{{ mktemp.stdout }}/metering-manifests/{{ item | basename }}"
+  with_fileglob:
+  - "*.yaml"
+
+- name: Update Metering Helm Operator deployment image
+  yedit:
+    src: "{{ mktemp.stdout }}/metering-manifests/metering-helm-operator-deployment.yaml"
+    edits:
+    - key: spec.template.spec.containers[0].image
+      value: "{{ openshift_metering_operator_image }}"
+    - key: spec.template.spec.containers[1].image
+      value: "{{ openshift_metering_operator_image }}"
+  when: openshift_metering_operator_image != ""
+
+- name: Update Metering config
+  yedit:
+    src: "{{ mktemp.stdout }}/metering-manifests/metering.yaml"
+    edits:
+    - key: spec
+      value: "{{ openshift_metering_config }}"
+  when: openshift_metering_config != None
+
+- name: "Create {{ __openshift_metering_namespace }} project"
+  oc_project:
+    state: present
+    name: "{{ __openshift_metering_namespace }}"
+    description: Openshift Metering
+    node_selector: ""
+
+- name: Create metering namespace viewer cluster role
+  oc_clusterrole:
+    state: present
+    name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+    rules:
+    - apiGroups: [""]
+      resources: ["namespaces"]
+      verbs: ["get"]
+
+- name: Create namespace viewer clusterrolebinding
+  oc_obj:
+    state: present
+    kind: ClusterRoleBinding
+    name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+    content:
+      path: "{{ mktemp.stdout }}/openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+      data:
+        apiVersion: v1
+        kind: ClusterRoleBinding
+        metadata:
+          name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: ClusterRole
+          name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+        subjects:
+        - kind: ServiceAccount
+          name: metering
+          namespace: "{{ __openshift_metering_namespace }}"
+
+- name: Install the Metering CRD
```

Comment:
Nitpick: `oc_obj` here would be easier, see ASB:

```
- name: Create custom resource definitions for asb
  oc_obj:
    name: '{{ crd.metadata.name }}'
    kind: CustomResourceDefinition
    state: present
    content:
      path: /tmp/{{ crd.metadata.name }}
      data: '{{ crd }}'
  vars:
    crd: "{{ lookup('file', item) | from_yaml }}"
  with_fileglob:
    - 'files/*.automationbroker.io.yaml'
```
It would automatically wait for it to be created. It would also obsolete copying admin kubeconfig and copying templates to the host

Reply:
We copy files cause we mutate some of them, but I can definitely switch to oc_obj here too. I gotta use two since I have to ensure the Metering CRD is created before the Metering CR.